ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLUtils.java

```diff
@@ -231,10 +231,17 @@ public static void validateTableIsIceberg(org.apache.hadoop.hive.ql.metadata.Tab
   }
 
   public static boolean isIcebergTable(Table table) {
-    return table.isNonNative() && 
-            table.getStorageHandler().getType() == StorageHandlerTypes.ICEBERG;
+    return table.isNonNative() &&
+            ((table.getStorageHandler() != null && table.getStorageHandler().getType() == StorageHandlerTypes.ICEBERG) || 
+                    isIcebergTableType(table.getTTable().getParameters()));
```

Comment:
Is there any case where StorageHandler#getType != ICEBERG but isIcebergTableType = true?

Reply:
no, shouldn't be

Reply:
Fixed @ rested on using MiniHS2Cluster for Iceberg and Hive ACID tables.